File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head == NULL)
        {
            return NULL;
        }

        ListNode *curr = head;
        int count = 0;
        while(curr != NULL)
        {
            count++;
            curr = curr->next;
        }

        if(n == count)
        {
            ListNode *head2 = head->next;
            delete head; 
            return head2;
        }

        int z = count - n;
        int count2 = 1; 
        ListNode *temp = head;
        while(temp != NULL)
        {
            if(count2 == z)
            {
                ListNode *curr2 = temp->next;
                temp->next = curr2->next;
                delete curr2; 
                break;
            }
            else
            {
                count2++;
                temp = temp->next;
            }
        }
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head == null){
            return null;
        }
        ListNode curr = head;
        int count = 0;
        while(curr!=null){
            count++;
            curr = curr.next;
        }
        if(n == count){
            return head.next;
        }
        int count2 = 0;
        ListNode temp = head;
        while(temp!=null){
            count2++;
            if(n == count - count2){
                
                ListNode temp2 = temp.next;
                temp.next = temp.next.next;
                temp2.next = null;
                return head;
            }
            temp = temp.next;
        }
        return null;
    }
}